controllers/appvolumereplication_controller.go

```diff
@@ -0,0 +1,377 @@
+		r.Client, placementRule.Spec.GenericPlacementFields, nil, placementRule)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get cluster map for placement %s error: %w", placementRule.Name, err)
+	}
+
+	if len(clmap) != clusterCount {
+		return "", "", fmt.Errorf("PlacementRule %s should have made 2 decisions. Found %d", placementRule.Name, len(clmap))
+	}
+
+	idx := 0
+
+	clusters := make([]spokeClusterV1.ManagedCluster, clusterCount)
+	for _, c := range clmap {
+		clusters[idx] = *c
+		idx++
+	}
+
+	d1 := clusters[0]
+	d2 := clusters[1]
+
+	var homeCluster string
+
+	var peerCluster string
+
+	switch {
+	case subStatuses[d1.Name] != nil:
+		homeCluster = d1.Name
+		peerCluster = d2.Name
+	case subStatuses[d2.Name] != nil:
+		homeCluster = d2.Name
+		peerCluster = d1.Name
+	default:
+		return "", "", fmt.Errorf("mismatch between placementRule %s decisions and subscription %s statuses",
+			placementRule.Name, subscription.Name)
+	}
+
+	return homeCluster, peerCluster, nil
+}
+
+func (r *AppVolumeReplicationReconciler) doCreateOrUpdateManifestWork(name string, namespace string,
+	homeCluster string, peerCluster string) error {
+	r.Log.Info("attempt to create and update ManifestWork")
+
+	manifestWork := r.createManifestWork(name, namespace, homeCluster, peerCluster)
+
+	found := &ocmworkv1.ManifestWork{}
+
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: manifestWork.Name, Namespace: homeCluster}, found)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			r.Log.Error(err, "Could not fetch ManifestWork")
+
+			return moreerrors.Wrap(err, "failed to fetch manifestWork")
+		}
+
+		r.Log.Info("Creating ManifestWork", "ManifestWork", manifestWork)
+
+		return r.Client.Create(context.TODO(), manifestWork)
+	} else if !reflect.DeepEqual(found.Spec, manifestWork.Spec) {
+		manifestWork.Spec.DeepCopyInto(&found.Spec)
+
+		r.Log.Info("Updating ManifestWork", "ManifestWork", manifestWork)
+
+		return r.Client.Update(context.TODO(), found)
+	}
+
+	return nil
+}
+
+func (r *AppVolumeReplicationReconciler) createManifestWork(name string, namespace string,
+	homeCluster string, peerCluster string) *ocmworkv1.ManifestWork {
+	vrgClientManifest := r.createVRGClientManifest(name, namespace, homeCluster, peerCluster)
+
+	manifestwork := &ocmworkv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf(manifestWorkNameFormat, homeCluster),
+			Namespace: homeCluster, Labels: map[string]string{"app": "VRG"},
+		},
+		Spec: ocmworkv1.ManifestWorkSpec{
+			Workload: ocmworkv1.ManifestsTemplate{
+				Manifests: []ocmworkv1.Manifest{
+					*vrgClientManifest,
+				},
+			},
+		},
+	}
+
+	return manifestwork
+}
+
+func (r *AppVolumeReplicationReconciler) createVRGClientManifest(name string, namespace string,
+	homeCluster string, peerCluster string) *ocmworkv1.Manifest {
+	// TODO: try to use the VRG object, then marshal it into byte array.
+	vrgClientJSON := []byte(fmt.Sprintf(`
+	{
```

Comment:
This is leveraging existing VRG, which needs some redefinition. Just noting this down, not intended to address as part of this PR.